packages/stubGenerator/languages/javascript/src/commandParsers/parseReadCommand.ts

```diff
@@ -0,0 +1,59 @@
+import * as R from 'ramda';
+import { match } from 'ts-pattern';
+
+import {
+  eliminateConstrains,
+  isAllOfType,
+  nameOf,
+  replaceLongsWithInt,
+  typeOf,
+  Variable,
+} from '../domain/';
+import { capitalize, throwErr } from '../utils';
+
+const declareVariable = ([varName, type]: string[], to: string) =>
+  match(type)
+    .with('float', () => `const ${varName} = toFloat(${to});`)
+    .with('int', () => `const ${varName} = toInt(${to});`)
+    .when(R.startsWith('string'), () => `const ${varName} = ${to};`)
+    .otherwise(() => throwErr(`variable type ${type} is unknown`));
+
+const parseVariablesOfDifferentType = (vars: Variable[]) => {
+  const nonWordsVars = vars.filter(R.pipe(typeOf, R.equals('word'), R.not));
+  const varsWithPlaceholder = vars.map(([name, type]): string =>
+    type === 'word' ? name : `raw${capitalize(name)}`,
+  );
+  return `const [${varsWithPlaceholder.join(', ')}] = readline().split(' ');
+    ${R.zipWith(
+      declareVariable,
+      nonWordsVars,
+      varsWithPlaceholder.filter(R.startsWith('raw')),
+    ).join('\n')}
```

Comment:
Should we have a name for these lines of code?
It's not readable.

Reply:
No, I mean this `zipWith` composition is not readable, it takes a lot of time to understand what is it doing. I prefer to extract that as a named function.